File: server/Models/HarvestChemical.cs
namespace Server.Models;

/// <summary>
/// A chemical (fertiliser, pesticide, treatment, …) applied to a <see cref="Harvest"/>, with the
/// date it was applied and what it cost. The cost is rolled into the harvest's final expenses.
/// Plain CRUD — applying a chemical moves no stock balance, like <see cref="HarvestTree"/>.
/// </summary>
public class HarvestChemical
{
    public int Id { get; set; }
    public int HarvestId { get; set; }

    /// <summary>Free-text name of the chemical applied.</summary>
    public string Name { get; set; } = string.Empty;

    public DateOnly Date { get; set; }

    /// <summary>What the chemical cost, added to the harvest's expense total.</summary>
    public decimal Cost { get; set; }
}

File: web/src/types/harvest-chemical.ts
/** A chemical applied to a harvest, with its cost — folded into the harvest's final expenses. */
export type HarvestChemical = {
  id: number;
  harvestId: number;
  name: string;
  /** ISO date string (YYYY-MM-DD). */
  date: string;
  cost: number;
};

export type HarvestChemicalInput = Omit<HarvestChemical, 'id'>;
